crypto/fipsmodule/modes/gcm.c

```diff
@@ -335,7 +337,7 @@ void CRYPTO_gcm128_setiv(GCM128_CONTEXT *ctx, const AES_KEY *key,
   ctx->ares = 0;
   ctx->mres = 0;
 
-#if defined(GHASH_ASM_X86_64)
+#if defined(GHASH_ASM_X86_64) && !defined(MY_ASSEMBLER_IS_TOO_OLD_FOR_512AVX)
```

Comment:
`GHASH_ASM_X86_64` itself is guarded here: https://github.com/aws/aws-lc/blob/88faaa28922a584b9df111cf2ead7f16fa6f6b41/crypto/fipsmodule/modes/internal.h#L296-L297